File: solutions/typescript/triangle/1/triangle.ts
export class Triangle {
  private sides: number[];

  constructor(...sides: number[]) {
    this.sides = sides.sort((a, b) => a - b);
  }

  private get isValid(): boolean {
    const [a, b, c] = this.sides;
    return a > 0 && a + b > c;
  }

  get isEquilateral(): boolean {
    if (!this.isValid) return false;
    const [a, b, c] = this.sides;
    return a === b && b === c;
  }

  get isIsosceles(): boolean {
    if (!this.isValid) return false;
    const [a, b, c] = this.sides;
    return a === b || b === c || a === c;
  }

  get isScalene(): boolean {
    if (!this.isValid) return false;
    const [a, b, c] = this.sides;

    return a !== b && b !== c && a !== c;
  }
}

File: solutions/julia/triangle/1/triangle.jl

function is_valid(sides)
    a, b, c = sides
    return a > 0 && b > 0 && c > 0 && a + b >= c && b + c >= a && a + c >= b
end


function is_equilateral(sides)
    if !is_valid(sides) return false end
    a, b, c = sides
    return a == b && b == c
end


function is_isosceles(sides)
    if !is_valid(sides) return false end
    a, b, c = sides
    return a == b || b == c || a == c
end


function is_scalene(sides)
    if !is_valid(sides) return false end
    a, b, c = sides
    return a != b && b != c && a != c
end
